### DISCUSSION TITLE
Haptics Web Implementation

### DISCUSSION THREAD

**Discussion by cameronrr** [ionic-team/capacitor#3347] [Ideas]
Are there any plans to add Web Support / Web Implementation for Haptics?

See as examples: http://www.hapticsjs.org/ (includes test examples) and https://shapeshed.com/html5-vibrate-api/

```javascript
// consolidate the different browser implementations as still candidate api
navigator.vibrate = navigator.vibrate || navigator.webkitVibrate || navigator.mozVibrate || navigator.msVibrate

// test whether vibration is supported by the current browser
if (navigator.vibrate) {
    console.log('Vibration is available')
} else {
    console.log('Vibration not available')
}

// single vibration. duration of vibration in ms
navigator.vibrate(500)

// vibration sequence. array sequence of vibration, rest, vibration ...
navigator.vibrate([500, 500, 500)
```

**Comment by imhoffd** [comment]
I think it would make sense as a feature for Capacitor 3. Would you mind creating a feature request in our new plugins repo? https://github.com/ionic-team/capacitor-plugins

Thanks!

**Reply by cameronrr** [reply]
Awesome, done! [https://github.com/ionic-team/capacitor-plugins/issues/4](https://github.com/ionic-team/capacitor-plugins/issues/4)

**Comment by jcesarmobile** [comment]
@capacitor/haptics 1.0.0 was released last week with web support
